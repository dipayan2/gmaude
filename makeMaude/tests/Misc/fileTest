#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < $srcdir/fileTest.maude -no-banner -no-advise -allow-files \
  > fileTest.out 2>&1

diff $srcdir/fileTest.expected fileTest.out > /dev/null 2>&1
