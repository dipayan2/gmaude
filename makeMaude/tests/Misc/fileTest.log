FAIL fileTest (exit status: 1)
