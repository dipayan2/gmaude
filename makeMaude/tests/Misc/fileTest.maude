***
***	Test file operations and error handling.
***

set show timing off .
set print conceal on .
print conceal file .

load file

mod FILE-TEST1 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op badName : -> String .
endm

erew <> < me : myClass | none > openFile(fileManager, me, badName, "r") .
erew <> < me : myClass | none > openFile(fileManager, me, "/shouldNotExist", "r") .


mod FILE-TEST2 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
  op badNat : -> Nat .
vars O O2 F : Oid .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "../../../tests/Misc/fileTest.maude", "r") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > getChars(F, O, badNat) .
endm

erew run .


mod FILE-TEST3 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
vars O O2 F : Oid .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "../../../tests/Misc/fileTest.maude", "r") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > getChars(F, O, 1000) .
endm

erew run .


mod FILE-TEST4 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
  op badNat : -> Nat .
vars O O2 F : Oid .
var S : String .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "../../../tests/Misc/fileTest.maude", "r") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > getChars(F, O, 1000) .

  rl < O : myClass | none > gotChars(O, F, S) =>
     < O : myClass | none > setPosition(F, O, badNat, start) .
endm

erew run .


mod FILE-TEST5 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
  op badNat : -> Nat .
vars O O2 F : Oid .
var S : String .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "../../../tests/Misc/fileTest.maude", "r") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > getChars(F, O, 1000) .

  rl < O : myClass | none > gotChars(O, F, S) =>
     < O : myClass | none > setPosition(F, O, badNat, current) .
endm

erew run .


mod FILE-TEST6 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
  op badNat : -> Nat .
vars O O2 F : Oid .
var S : String .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "../../../tests/Misc/fileTest.maude", "r") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > getChars(F, O, 1000) .

  rl < O : myClass | none > gotChars(O, F, S) =>
     < O : myClass | none > setPosition(F, O, badNat, end) .
endm

erew run .


mod FILE-TEST7 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
  op badBase : -> Base .
vars O O2 F : Oid .
var S : String .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "../../../tests/Misc/fileTest.maude", "r") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > getChars(F, O, 1000) .

  rl < O : myClass | none > gotChars(O, F, S) =>
     < O : myClass | none > setPosition(F, O, 100, badBase) .
endm

erew run .


mod FILE-TEST8 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
  op state : Nat -> Attribute .
vars O O2 F : Oid .
var S : String .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "../../../tests/Misc/fileTest.maude", "r") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > getChars(F, O, 1000) .

  rl < O : myClass | none > gotChars(O, F, S) =>
     < O : myClass | none > setPosition(F, O, 100, start) .

  rl < O : myClass | none > positionSet(O, F) =>
     < O : myClass | state(1) > getChars(F, O, 100) .
endm

erew run .


mod FILE-TEST9 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
  op state : Nat -> Attribute .
vars O O2 F : Oid .
var S : String .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "../../../tests/Misc/fileTest.maude", "r") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > getChars(F, O, 1000) .

  rl < O : myClass | none > gotChars(O, F, S) =>
     < O : myClass | none > setPosition(F, O, 100, start) .

  rl < O : myClass | none > positionSet(O, F) =>
     < O : myClass | state(1) > getChars(F, O, 100) .

  rl < O : myClass | state(1) > gotChars(O, F, S) =>
     < O : myClass | state(1) > getPosition(F, O) .

endm

erew run .


mod FILE-TEST10 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
vars O O2 F : Oid .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "../../../tests/Misc/fileTest.maude", "r") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > write(F, O, "test") .
endm

erew run .


mod FILE-TEST11 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
vars O O2 F : Oid .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "../../../tests/Misc/fileTest.maude", "r") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > flush(F, O) .
endm

erew run .


mod FILE-TEST12 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
vars O O2 F : Oid .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "fileTest.tmp", "w") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > write(F, O, "testing testing 1 2 3\n") .

  rl < O : myClass | none > wrote(O, F) =>
     < O : myClass | none > getChars(F, O, 100) .
endm

erew run .


mod FILE-TEST13 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
vars O O2 F : Oid .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "fileTest.tmp", "w") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > write(F, O, "testing testing 1 2 3\n") .

  rl < O : myClass | none > wrote(O, F) =>
     < O : myClass | none > getLine(F, O) .
endm

erew run .


mod FILE-TEST14 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
vars O O2 F : Oid .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "fileTest.tmp", "w+") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > write(F, O, "testing testing 1 2 3\n") .

  rl < O : myClass | none > wrote(O, F) =>
     < O : myClass | none > setPosition(F, O, 0, start) .

  rl < O : myClass | none > positionSet(O, F) =>
     < O : myClass | none > getLine(F, O) .
endm

erew run .


mod FILE-TEST15 is
  pr FILE .
  op myClass : -> Cid .
  ops me : -> Oid .
  op run : -> Configuration .
  op badName : -> String .

vars O O2 F : Oid .
var S : String .

  eq run = <> < me : myClass | none > openFile(fileManager, me, "fileTest.tmp", "w+") .

  rl < O : myClass | none > openedFile(O, O2, F) =>
     < O : myClass | none > write(F, O, "testing testing 1 2 3\n") .

  rl < O : myClass | none > wrote(O, F) =>
     < O : myClass | none > setPosition(F, O, 0, start) .

  rl < O : myClass | none > positionSet(O, F) =>
     < O : myClass | none > getLine(F, O) .

  rl < O : myClass | none > gotLine(O, F, S) =>
     < O : myClass | none > closeFile(F, O) .
  
endm

erew run .
erew <> < me : myClass | none > removeFile(fileManager, me, "fileTest.tmp") .
erew <> < me : myClass | none > removeFile(fileManager, me, badName) .
erew <> < me : myClass | none > removeFile(fileManager, me, "fileTest.tmp") .
