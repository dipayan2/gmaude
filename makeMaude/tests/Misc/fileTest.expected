==========================================
erewrite in FILE-TEST1 : <> < me : myClass | none > openFile(fileManager, me,
    badName, "r") .
rewrites: 0
result Configuration: <> < me : myClass | none > fileError(me, fileManager,
    "Bad file name.")
==========================================
erewrite in FILE-TEST1 : <> < me : myClass | none > openFile(fileManager, me,
    "/shouldNotExist", "r") .
rewrites: 0
result Configuration: <> < me : myClass | none > fileError(me, fileManager,
    "No such file or directory")
==========================================
erewrite in FILE-TEST2 : run .
rewrites: 2
result Configuration: <> < me : myClass | none > fileError(me, file(...),
    "Bad size.")
==========================================
erewrite in FILE-TEST3 : run .
rewrites: 2
result Configuration: <> < me : myClass | none > gotChars(me, file(...), "***\n***\tTest file operations and error handling.\n***\n\nset show timing off .\nset print conceal on .\nprint conceal file .\n\nload file\n\nmod FILE-TEST1 is\n  pr FILE .\n  op myClass : -> Cid .\n  ops me : -> Oid .\n  op badName : -> String .\nendm\n\nerew <> < me : myClass | none > openFile(fileManager, me, badName, \"r\") .\nerew <> < me : myClass | none > openFile(fileManager, me, \"/shouldNotExist\", \"r\") .\n\n\nmod FILE-TEST2 is\n  pr FILE .\n  op myClass : -> Cid .\n  ops me : -> Oid .\n  op run : -> Configuration .\n  op badNat : -> Nat .\nvars O O2 F : Oid .\n\n  eq run = <> < me : myClass | none > openFile(fileManager, me, \"../../../tests/Misc/fileTest.maude\", \"r\") .\n\n  rl < O : myClass | none > openedFile(O, O2, F) =>\n     < O : myClass | none > getChars(F, O, badNat) .\nendm\n\nerew run .\n\n\nmod FILE-TEST3 is\n  pr FILE .\n  op myClass : -> Cid .\n  ops me : -> Oid .\n  op run : -> Configuration .\nvars O O2 F : Oid .\n\n  eq run = <> < me : myClass | none > openFile(fileManager, me, \"../../../tests/Misc/fileTest.")
==========================================
erewrite in FILE-TEST4 : run .
rewrites: 3
result Configuration: <> < me : myClass | none > fileError(me, file(...),
    "Bad offset.")
==========================================
erewrite in FILE-TEST5 : run .
rewrites: 3
result Configuration: <> < me : myClass | none > fileError(me, file(...),
    "Bad offset.")
==========================================
erewrite in FILE-TEST6 : run .
rewrites: 3
result Configuration: <> < me : myClass | none > fileError(me, file(...),
    "Bad offset.")
==========================================
erewrite in FILE-TEST7 : run .
rewrites: 3
result Configuration: <> < me : myClass | none > fileError(me, file(...),
    "Bad base.")
==========================================
erewrite in FILE-TEST8 : run .
rewrites: 4
result Configuration: <> < me : myClass | state(1) > gotChars(me, file(...), "rint conceal file .\n\nload file\n\nmod FILE-TEST1 is\n  pr FILE .\n  op myClass : -> Cid .\n  ops me : -> ")
==========================================
erewrite in FILE-TEST9 : run .
rewrites: 5
result Configuration: <> < me : myClass | state(1) > positionGot(me, file(...),
    200)
==========================================
erewrite in FILE-TEST10 : run .
rewrites: 2
result Configuration: <> < me : myClass | none > fileError(me, file(...),
    "File not open for writing.")
==========================================
erewrite in FILE-TEST11 : run .
rewrites: 2
result Configuration: <> < me : myClass | none > fileError(me, file(...),
    "File not open for writing.")
==========================================
erewrite in FILE-TEST12 : run .
rewrites: 3
result Configuration: <> < me : myClass | none > fileError(me, file(...),
    "File not open for reading.")
==========================================
erewrite in FILE-TEST13 : run .
rewrites: 3
result Configuration: <> < me : myClass | none > fileError(me, file(...),
    "File not open for reading.")
==========================================
erewrite in FILE-TEST14 : run .
rewrites: 4
result Configuration: <> < me : myClass | none > gotLine(me, file(...),
    "testing testing 1 2 3\n")
==========================================
erewrite in FILE-TEST15 : run .
rewrites: 5
result Configuration: <> closedFile(me, file(...)) < me : myClass | none >
==========================================
erewrite in FILE-TEST15 : <> < me : myClass | none > removeFile(fileManager,
    me, "fileTest.tmp") .
rewrites: 0
result Configuration: <> removedFile(me, fileManager) < me : myClass | none >
==========================================
erewrite in FILE-TEST15 : <> < me : myClass | none > removeFile(fileManager,
    me, badName) .
rewrites: 0
result Configuration: <> < me : myClass | none > fileError(me, fileManager,
    "Bad file name.")
==========================================
erewrite in FILE-TEST15 : <> < me : myClass | none > removeFile(fileManager,
    me, "fileTest.tmp") .
rewrites: 0
result Configuration: <> < me : myClass | none > fileError(me, fileManager,
    "No such file or directory")
Bye.
